api/python/t4/packages.py

```diff
@@ -230,6 +230,18 @@ def __init__(self):
         self._children = {}
         self._meta = {'version': 'v0'}
 
+    def __repr__(self, level=0, indent='  '):
+        """
+        String representation of the Package.
+        """
+        self_repr = ''
+        for child_key in sorted(self.keys()):
+            child_entry = indent*level + child_key + '\n'
+            self_repr += child_entry
+            if isinstance(self[child_key], Package):
+                self_repr += self[child_key].__repr__(level+1, indent)
```

Comment:
You could save yourself an argument by just appending to indent when making the recursive call (no need for level then). Strings are immutable in python so that won't impact the string in the calling frame
```
self_repr += self[child_key].__repr__(constants.INDENT + indent)
```
No strong preference, just pointing out an option.

Reply:
I like having it as an argument in case someone wants to use a different indent. It's a very small consideration, but it's why I made it an argument instead of a constant.